### ISSUE TITLE
Songs from one album get split into multiple albums based on artist names

### ISSUE THREAD

**Issue by Deab22** [moritz-weber/mucke#155]
Hey, i really like this music player app, thanks for making it!

i have a small issue tho, relating to albums with collaborations.

Currently, if you load an album where some of the songs are collaborations, these songs are put into their own albums, with identical names and covers

![Screenshot_20231024_202012](https://github.com/moritz-weber/mucke/assets/88623347/c66aa437-b018-4d5e-a1f7-fdb6a34631fc)
(the second album in the screenshot contains only one song...)

Usually, I'd expect these songs to be in one album, not split up
is this behavior intentional?

if not, you might be able to fix this by comparing album names and/or covers, and adding the artist names of the albums together

(I'd like to fix this myself and make a pull request, but I've never written a single line of dart code in my life, sry...)

Version 1.5.1 from F-Droid
Samsung Galaxy S10
Android 12

**Comment by moritz-weber** [comment]
Thank you for the feedback. Right now, albums are identified by year, name and album artist. If there is no album artist found in the metadata, the artist is used instead.

This works pretty well in most cases and feels like a good approach to me. Removing the artist as a criterion would probably work out fine in most cases, but the risk of collisions is definitely higher.

As an easy fix, I would suggest adding the "album artist" tag to the albums that require this info (or just to all songs).

With this, the collaborations are no issue:
![Screenshot_20231024-201028.png](https://github.com/moritz-weber/mucke/assets/17176771/52ca9db8-28b6-4b86-82bc-e3b93f8515e0)

Edit: There is definitely a style issue to fix in this screenshot 😅

**Comment by ildar** [comment]
MusicBrainz Album id isn't considered?

**Comment by Deab22** [comment]
Thanks for your answer!

i didn't know about the album artist tag, makes sense

aparrently, the app i used to get the tags has the problem that it sets the album artist to the song artist name no matter what, so it's an external issue

**Comment by moritz-weber** [comment]
@ildar No. I don't think that there is an interface to read this tag currently. And it's also not really a standard tag.

@Deab22 MusicBrainz Picard works pretty well for me for tagging.

**Comment by ildar** [comment]
ср, 25 окт. 2023 г., 18:52 Moritz Weber:

> And it's also not really a standard tag.
>
There isn't any standard for this. That's why Music Brainz became de facto
standard.
Many people use it including you ))
